### PR TITLE
Ignore pipeline default solutions as best a priori WCS

### PR DESCRIPTION
**Pull request by stsci-hack**
New WCS solutions from the astrometry database includes pipeline default WCS solutions as a record of the calibration of the astrometry of the observation.  Unfortunately, sometimes these are the only solutions for an observation in the database, and therefore they get flagged as the 'best' solution for the exposure.  This, however, will overwrite the distortion correction being applied by 'updatewcs' which most likely is what the user really wants.  

This revision allows all the unique solutions from the astrometry database to still be appended as headerlet extensions to the exposure, but does NOT apply as the primary WCS any pipeline default solution (either OPUS or IDC_* WCS) from the database.

### CODE REVIEW

**Comment by stsci-hack** [comment]
This issue came as a result of testing for HSTDP-2019.3a with the processing of exposure j9i413bvq.

File: stwcs/updatewcs/astrometry_utils.py [hunk 1]
```diff
@@ -158,6 +158,10 @@ def updateObs(self, obsname):
         wcsnames = headerlet.get_headerlet_kw_names(obsname, 'wcsname')
 
         headerlets, best_solution_id = self.getObservation(observationID)
+        # Determine whether best_solution_id is simply the pipeline default solution
+        if len(best_solution_id.split('-')) == 1: 
+            best_solution_id = None
```
Comment (nden) [inline stwcs/updatewcs/astrometry_utils.py]: How does this condition actually determine that it is the default pipeline solution?
This implies some naming convention which I am not familiar with.

Comment (nden) [reply stwcs/updatewcs/astrometry_utils.py]: And what is the "default pipeline" solution? Is it the original OPUS WCS?

File: CHANGES.rst [hunk 2]
```diff
@@ -1,6 +1,8 @@
 1.5.1 (Unreleased)
 ------------------
 
+- Do not apply pipeline-default WCSs from astrometry database as new WCS. [#96]
+
```
Comment (nden) [inline CHANGES.rst]: Is this going to DMS now? If not should it go into 1.5.2?

File: stwcs/updatewcs/astrometry_utils.py (hunk 1 above)
Comment (stsci-hack) [reply stwcs/updatewcs/astrometry_utils.py]: The astrometry database will return an empty string (or None) as the value when no solution has been identified as the 'best' solution.  Also, the WCSNAME for all a priori and a posteriori solutions will start with the 'pipeline default' solutions (WCSNAME=OPUS or IDC_<ipppssoot>) and append the new solution information to them joined by '-'. This is documented in the Drizzlepac readthedocs section on the astrometry solutions (https://drizzlepac.readthedocs.io/en/latest/astrometry.html#interpreting-wcs-names) and in the public pages for this project (https://outerspace.stsci.edu/display/HAP).   

Should this condition be met, it means there is no a priori or archive a posteriori solutions in the database.  In those cases, all solutions from the database will be appended to the exposure being processed and the current distortion model which updatewcs just applied (which I referred to as the 'pipeline default') will remain unchanged as the 'best' solution for subsequent processing.  
I have also accounted for empty return value from the database in commit 4ed7a56.

File: CHANGES.rst (hunk 2 above)
Comment (stsci-hack) [reply CHANGES.rst]: We do not have any reason now for this change to go to DMS now (something I was not sure of when I first submitted this PR), so I changed this to a comment on 1.5.2 in commit 6c60cf6.

**Comment by nden** [comment]
@stsci-hack Is this still relevant?

**Comment by stsci-hack** [comment]
This should not be necessary any more due to changes in the astrometry database in assigning 'best' wcs.

**Pull request closed without merging**